src/main/scala/io/iohk/ethereum/merklepatriciatrie/Node.scala

```diff
@@ -0,0 +1,102 @@
+package io.iohk.ethereum.merklepatriciatrie
+
+import io.iohk.ethereum.rlp.{decode => decodeRLP, encode => encodeRLP, _}
+import io.iohk.ethereum.merklepatriciatrie.MerklePatriciaTrie.HashFn
+
+/**
+  * Trie elements
+  */
+private[merklepatriciatrie] sealed trait Node {
+
+  import MerklePatriciaTrie._
+
+  private var hashedCache: Option[Array[Byte]] = None
```

Comment:
if `hashFn` should not change maybe we can add it as `val` in `Node` trait?
this way it should be easier to introduce lazy val